### PR TITLE
Refactoring/#258 used a non root user to run jupyter in the docker image ai lab

### LINKED ISSUE #66: Use a non-root user to run Jupyter in the Docker Image
Security best practices recommend not to run services in Docker with root. We should prepare a non-root user which is used to run Jupyter.

https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-2-set-a-user

In total we will need two users besides `root`
* User with `sudo` permission  used for Ansible installation using `sudo` where required.
  * name `ansible` for Docker Edition
  * name `ubuntu` AMI and VM images
* User `jupyter` without `sudo` permission useds for running the docker container and Jupyterlab.

### LINKED ISSUE #258: te_init: Explain where the container activation happens.
From a ai-lab review:

_"When the initialization finishes we should see a printed output suggesting us to activate the language container. Let's take it as an indication of the successful completion of the initialization procedure. We are not going to do the language container activation manually."_

So who is doing the language container activation? The script is doing it automatically?

### PR DESCRIPTION
**Pull request by ckunki**
Closes #66

### CODE REVIEW

File: exasol/ds/sandbox/lib/dss_docker/create_image.py [hunk 1]
```diff
@@ -113,6 +113,10 @@ def _docker_file(self) -> importlib_resources.abc.Traversable:
     def _start_container(self) -> DockerContainer:
         self._start = datetime.now()
         docker_client = docker.from_env()
+        try:
```
Comment (ckunki) [inline exasol/ds/sandbox/lib/dss_docker/create_image.py:116]: I added these lines to speed up testing Ansible tasks

File: exasol/ds/sandbox/lib/dss_docker/create_image.py [hunk 2]
```diff
@@ -113,6 +113,10 @@ def _docker_file(self) -> importlib_resources.abc.Traversable:
...
+            return docker_client.containers.get(self.container_name)
+        except:
+            pass
```
Comment (ckunki) [inline exasol/ds/sandbox/lib/dss_docker/create_image.py:119]: I added these lines to speed up testing Ansible tasks

File: exasol/ds/sandbox/runtime/ansible/general_setup_tasks.yml [hunk 3]
```diff
@@ -36,7 +34,16 @@
...
+- name: Change owner of all files and dirs in home directory
+  ansible.builtin.file:
+    path: "{{ user_home }}"
+    owner: "{{ user_name }}"
+    group: "{{ user_group }}"
+    recurse: true
+  become: "{{ need_sudo }}"
 - name: Clear pip cache
+  # TBD: do we need become: "{{ need_sudo }}" here?
+  # should folder /root be replaced by {{ user_home }} ?
```
Comment (ckunki) [inline exasol/ds/sandbox/runtime/ansible/general_setup_tasks.yml:46]: Currently I did not find such a directory at all.

File: exasol/ds/sandbox/runtime/ansible/roles/jupyter/tasks/pip-install.yml
Comment (ckunki) [inline exasol/ds/sandbox/runtime/ansible/roles/jupyter/tasks/pip-install.yml:1]: This file originally was named `install-pip-packages.yml`.

Additionally, I renamed some of the variables in this file and fixed an Ansible warning about 
providing a list of dictionaries to vars: See https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_8.html#deprecated

**Review by tkilias — CHANGES_REQUESTED** [review]

File: test/codebuild/test_ci.py [hunk 4]
```diff
@@ -94,6 +94,7 @@ def new_ec2_from_ami():
                 status = aws_access.get_instance_status(ec2_instance_description.id)
             assert status.ok
             time.sleep(10)
+            # probably we need to set the password for user "jupyter" here.
```
Comment (tkilias) [inline test/codebuild/test_ci.py:97]: Not necessary anymore

File: test/integration/test_create_dss_docker_image.py [hunk 5]
```diff
@@ -44,6 +45,8 @@ def retry(exception: typing.Type[BaseException], timeout: timedelta):
     )
 
 
+@pytest.mark.skipif(os.environ.get('DSS_RUN_CI_TEST') != 'true',
```
Comment (tkilias) [inline test/integration/test_create_dss_docker_image.py:48]: Why is this needed?

File: test/integration/test_motd_jupyter_template.py [hunk 6]
```diff
@@ -11,7 +11,8 @@
 
 @pytest.fixture()
 def motd_file(tmp_path):
-    jupyter_server_config_file = tmp_path / "jupyter_server_config.json"
+    # jupyterlab_config_json
```
Comment (tkilias) [inline test/integration/test_motd_jupyter_template.py:14]: Comment necessary

**Review by tkilias — CHANGES_REQUESTED** [review]

File: exasol/ds/sandbox/runtime/ansible/roles/jupyter/templates/etc/update-motd.d/999-jupyter [hunk 7]
```diff
@@ -11,11 +11,11 @@ You can connect with https://<host>:{{jupyterlab_port}}.
 {{heading_jupyter_update_password}}
 
 The default password is "{{jupyterlab_password}}".
-To update the password as user {{user_name}} run
+To update the password as user {{ user_name }} run
```
Comment (tkilias) [inline exasol/ds/sandbox/runtime/ansible/roles/jupyter/templates/etc/update-motd.d/999-jupyter:14]:
```suggestion
To update the password run as user {{ user_name }} the following command
```

or should we directly show it with sudo

**Review by tkilias — APPROVED** [review]

**Pull request merged**